SimpleProducer.py

```diff
@@ -53,18 +56,25 @@ def send(server_args, producer_counters, topic, shared_data_queue, avro_schema,
             val = shared_data_queue.get()
             if val is None:
                 break
-            producer.send(topic, serialize_val(val, serializer, schema)).add_callback(on_send_success, producer_counters)
+            producer.send(topic, serialize_val(val, serializer, schema)).add_callback(on_send_success, producer_counters).add_errback(on_send_error, producer_counters)
 
 def on_send_success(producer_counters, _):
     producer_counters.received_counter.increment()
 
+def on_send_error(producer_counters, _):
```

Comment:
Should these be private? I don't know if it's worth it or not but in theory these should not be accessible from other files right?

Reply:
This would seem like a lot of the main producer code should be privatised under this ideology. I don't think its too much of a concern for the producer as it was for the classes generated for the datagen.